From 23ad766d2ecbeebbe9d5da610ae5f17e2df19564 Mon Sep 17 00:00:00 2001
From: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Date: Tue, 2 Aug 2022 15:50:48 +0300
Subject: [PATCH 25/42] dts: s32*: switch to generic pinconf/pinmux interface

Switch all pinctrl definitions to use the generic pinconf/pinmux
interface.

Issue: ALB-8868

Upstream-Status: Pending 

Signed-off-by: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 fdts/s32g.dtsi             | 395 +++++++++++++++++++++-----------
 fdts/s32g274a-bluebox3.dts | 328 +++++++++++++++++++--------
 fdts/s32g274a-emu.dts      |  25 ++-
 fdts/s32g399a-emu.dts      |  25 ++-
 fdts/s32gxxxa-evb.dtsi     | 378 +++++++++++++++++++++----------
 fdts/s32gxxxa-rdb.dtsi     | 322 ++++++++++++++++++--------
 fdts/s32r45-evb.dts        | 450 +++++++++++++++++++++++++------------
 7 files changed, 1323 insertions(+), 600 deletions(-)

diff --git a/fdts/s32g.dtsi b/fdts/s32g.dtsi
index d9ecc9f34..909aa381b 100644
--- a/fdts/s32g.dtsi
+++ b/fdts/s32g.dtsi
@@ -336,169 +336,306 @@
 };
 
 &pinctrl {
-	board_generic_pinctrl {
-		u-boot,dm-pre-reloc;
+	u-boot,dm-pre-reloc;
 	sd0_pins: sd0 {
-		sd0_grp {
-			fsl,pins = <PC14_MSCR_S32G PC14_SD0_CLK_CFG
-				    PC15_MSCR_S32G PC15_SD0_CMD_CFG
-				    PD00_MSCR_S32G PD00_SD0_D0_CFG
-				    PD01_MSCR_S32G PD01_SD0_D1_CFG
-				    PD02_MSCR_S32G PD02_SD0_D2_CFG
-				    PD03_MSCR_S32G PD03_SD0_D3_CFG
-				    PD04_MSCR_S32G PD04_SD0_D4_CFG
-				    PD05_MSCR_S32G PD05_SD0_D5_CFG
-				    PD06_MSCR_S32G PD06_SD0_D6_CFG
-				    PD07_MSCR_S32G PD07_SD0_D7_CFG
-				    PD08_MSCR_S32G PD08_SD0_RST_CFG
-				    PD09_MSCR_S32G PD09_SD0_VSELECT_CFG
-				    PD10_MSCR_S32G PD10_SD0_DQS_CFG
-				    SD0_CMD_IMCR PC15_SD0_CMD_IN
-				    SD0_D0_IMCR PD00_SD0_D0_IN
-				    SD0_D1_IMCR PD01_SD0_D1_IN
-				    SD0_D2_IMCR PD02_SD0_D2_IN
-				    SD0_D3_IMCR PD03_SD0_D3_IN
-				    SD0_D4_IMCR PD04_SD0_D4_IN
-				    SD0_D5_IMCR PD05_SD0_D5_IN
-				    SD0_D6_IMCR PD06_SD0_D6_IN
-				    SD0_D7_IMCR PD07_SD0_D7_IN
-				    SD0_DQS_IMCR PD10_SD0_DQS_IN
-				    >;
+		sd0_grp0 {
+			pinmux = <S32CC_PINMUX(46, FUNC1)>,
+				 <S32CC_PINMUX(56, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-down;
 		};
+
+		sd0_grp1 {
+			pinmux = <S32CC_PINMUX(47, FUNC1)>,
+				 <S32CC_PINMUX(48, FUNC1)>,
+				 <S32CC_PINMUX(49, FUNC1)>,
+				 <S32CC_PINMUX(50, FUNC1)>,
+				 <S32CC_PINMUX(51, FUNC1)>,
+				 <S32CC_PINMUX(52, FUNC1)>,
+				 <S32CC_PINMUX(53, FUNC1)>,
+				 <S32CC_PINMUX(54, FUNC1)>,
+				 <S32CC_PINMUX(55, FUNC1)>;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-up;
+		};
+
+		sd0_grp2 {
+			pinmux = <S32CC_PINMUX(57, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		sd0_grp3 {
+			pinmux = <S32CC_PINMUX(58, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		sd0_grp4 {
+			pinmux = <S32CC_PINMUX(515, FUNC2)>,
+				 <S32CC_PINMUX(516, FUNC2)>,
+				 <S32CC_PINMUX(517, FUNC2)>,
+				 <S32CC_PINMUX(520, FUNC2)>,
+				 <S32CC_PINMUX(521, FUNC2)>,
+				 <S32CC_PINMUX(522, FUNC2)>,
+				 <S32CC_PINMUX(523, FUNC2)>,
+				 <S32CC_PINMUX(519, FUNC2)>,
+				 <S32CC_PINMUX(518, FUNC2)>,
+				 <S32CC_PINMUX(524, FUNC2)>;
+		};
+
 	};
 
 	gmac0_mdio_pins: gmac0_mdio {
-		gmac0_mdio_grp {
-			fsl,pins = <PD12_MSCR_S32G PD12_GMAC0_MDC_CFG
-				    PD13_MSCR_S32G PD13_GMAC0_MDIO_CFG
-				    GMAC0_MDIO_IMCR PD13_GMAC0_MDIO_IN
-				    >;
+		gmac0_mdio_grp0 {
+			pinmux = <S32CC_PINMUX(60, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		gmac0_mdio_grp1 {
+			pinmux = <S32CC_PINMUX(61, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
 		};
+
+		gmac0_mdio_grp2 {
+			pinmux = <S32CC_PINMUX(527, FUNC2)>;
+		};
+
 	};
 
 	gmac0_pins: gmac0 {
-		gmac0_grp {
-			fsl,pins = <PE02_MSCR_S32G PE02_GMAC0_TX_CLK_CFG
-				    PE03_MSCR_S32G PE03_GMAC0_TX_EN_CFG
-				    PE04_MSCR_S32G PE04_GMAC0_TX_D0_CFG
-				    PE05_MSCR_S32G PE05_GMAC0_TX_D1_CFG
-				    PE06_MSCR_S32G PE06_GMAC0_TX_D2_CFG
-				    PE07_MSCR_S32G PE07_GMAC0_TX_D3_CFG
-				    PE08_MSCR_S32G PE08_GMAC0_RX_CLK_CFG
-				    PE09_MSCR_S32G PE09_GMAC0_RX_DV_CFG
-				    PE10_MSCR_S32G PE10_GMAC0_RX_D0_CFG
-				    PE11_MSCR_S32G PE11_GMAC0_RX_D1_CFG
-				    PE12_MSCR_S32G PE12_GMAC0_RX_D2_CFG
-				    PE13_MSCR_S32G PE13_GMAC0_RX_D3_CFG
-				    GMAC0_TX_CLK_IMCR PE02_GMAC0_TX_CLK_IN
-				    GMAC0_RX_CLK_IMCR PE08_GMAC0_RX_CLK_IN
-				    GMAC0_RX_DV_IMCR PE09_GMAC0_RX_DV_IN
-				    GMAC0_RX_D0_IMCR PE10_GMAC0_RX_D0_IN
-				    GMAC0_RX_D1_IMCR PE11_GMAC0_RX_D1_IN
-				    GMAC0_RX_D2_IMCR PE12_GMAC0_RX_D2_IN
-				    GMAC0_RX_D3_IMCR PE13_GMAC0_RX_D3_IN
-				    >;
+		gmac0_grp0 {
+			pinmux = <S32CC_PINMUX(66, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-up;
+		};
+
+		gmac0_grp1 {
+			pinmux = <S32CC_PINMUX(67, FUNC1)>,
+				 <S32CC_PINMUX(68, FUNC1)>,
+				 <S32CC_PINMUX(69, FUNC1)>,
+				 <S32CC_PINMUX(70, FUNC1)>,
+				 <S32CC_PINMUX(71, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		gmac0_grp2 {
+			pinmux = <S32CC_PINMUX(72, FUNC0)>,
+				 <S32CC_PINMUX(73, FUNC0)>,
+				 <S32CC_PINMUX(74, FUNC0)>,
+				 <S32CC_PINMUX(75, FUNC0)>,
+				 <S32CC_PINMUX(76, FUNC0)>,
+				 <S32CC_PINMUX(77, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		gmac0_grp3 {
+			pinmux = <S32CC_PINMUX(538, FUNC2)>,
+				 <S32CC_PINMUX(529, FUNC2)>,
+				 <S32CC_PINMUX(530, FUNC2)>,
+				 <S32CC_PINMUX(531, FUNC2)>,
+				 <S32CC_PINMUX(532, FUNC2)>,
+				 <S32CC_PINMUX(533, FUNC2)>,
+				 <S32CC_PINMUX(534, FUNC2)>;
 		};
+
 	};
 
 	pfe0_mdio_pins: pfe0_mdio {
-		pfe0_mdio_grp {
-			fsl,pins = <PE15_MSCR_S32G PE15_PFE0_MDIO_CFG
-				    PF02_MSCR_S32G PF02_PFE0_MDC_CFG
-				    PFE0_MDIO_IMCR PE15_PFE0_MDIO_IN
-				    >;
+		pfe0_mdio_grp0 {
+			pinmux = <S32CC_PINMUX(79, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		pfe0_mdio_grp1 {
+			pinmux = <S32CC_PINMUX(82, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
 		};
+
+		pfe0_mdio_grp2 {
+			pinmux = <S32CC_PINMUX(837, FUNC2)>;
+		};
+
 	};
 
 	pfe0_pins: pfe0 {
-		pfe0_grp {
-			fsl,pins = <PE14_MSCR_S32G PE14_PFE0_TX_EN_CFG
-				    PH01_MSCR_S32G PH01_PFE0_TX_D1_CFG
-				    PH02_MSCR_S32G PH02_PFE0_TX_D2_CFG
-				    PH03_MSCR_S32G PH03_PFE0_TX_D3_CFG
-				    PH04_MSCR_S32G PH04_PFE0_RX_CLK_CFG
-				    PH05_MSCR_S32G PH05_PFE0_RX_DV_CFG
-				    PH06_MSCR_S32G PH06_PFE0_RX_D0_CFG
-				    PH07_MSCR_S32G PH07_PFE0_RX_D1_CFG
-				    PH08_MSCR_S32G PH08_PFE0_RX_D2_CFG
-				    PH09_MSCR_S32G PH09_PFE0_RX_D3_CFG
-				    PH10_MSCR_S32G PH10_PFE0_TX_CLK_CFG
-				    PJ00_MSCR_S32G PJ00_PFE0_TX_D0_CFG
-				    PFE0_RX_CLK_IMCR PH04_PFE0_RX_CLK_IN
-				    PFE0_RX_DV_IMCR PH05_PFE0_RX_DV_IN
-				    PFE0_RX_D0_IMCR PH06_PFE0_RX_D0_IN
-				    PFE0_RX_D1_IMCR PH07_PFE0_RX_D1_IN
-				    PFE0_RX_D2_IMCR PH08_PFE0_RX_D2_IN
-				    PFE0_RX_D3_IMCR PH09_PFE0_RX_D3_IN
-				    >;
+		pfe0_grp0 {
+			pinmux = <S32CC_PINMUX(78, FUNC1)>,
+				 <S32CC_PINMUX(113, FUNC1)>,
+				 <S32CC_PINMUX(114, FUNC1)>,
+				 <S32CC_PINMUX(115, FUNC1)>,
+				 <S32CC_PINMUX(144, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		pfe0_grp1 {
+			pinmux = <S32CC_PINMUX(116, FUNC0)>,
+				 <S32CC_PINMUX(117, FUNC0)>,
+				 <S32CC_PINMUX(118, FUNC0)>,
+				 <S32CC_PINMUX(119, FUNC0)>,
+				 <S32CC_PINMUX(120, FUNC0)>,
+				 <S32CC_PINMUX(121, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		pfe0_grp2 {
+			pinmux = <S32CC_PINMUX(122, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-up;
 		};
+
+		pfe0_grp3 {
+			pinmux = <S32CC_PINMUX(839, FUNC2)>,
+				 <S32CC_PINMUX(845, FUNC2)>,
+				 <S32CC_PINMUX(841, FUNC2)>,
+				 <S32CC_PINMUX(842, FUNC2)>,
+				 <S32CC_PINMUX(843, FUNC2)>,
+				 <S32CC_PINMUX(844, FUNC2)>;
+		};
+
 	};
 
 	pfe1_mdio_pins: pfe1_mdio {
-		pfe1_mdio_grp {
-			fsl,pins = <PD12_MSCR_S32G PD12_PFE1_MDC_CFG
-				    PD13_MSCR_S32G PD13_PFE1_MDIO_CFG
-				    PFE1_MDIO_IMCR PD13_PFE1_MDIO_IN
-				    >;
+		pfe1_mdio_grp0 {
+			pinmux = <S32CC_PINMUX(60, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		pfe1_mdio_grp1 {
+			pinmux = <S32CC_PINMUX(61, FUNC2)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		pfe1_mdio_grp2 {
+			pinmux = <S32CC_PINMUX(857, FUNC2)>;
 		};
+
 	};
 
 	pfe1_pins: pfe1 {
-		pfe1_grp {
-			fsl,pins = <PE02_MSCR_S32G PE02_PFE1_TX_CLK_CFG
-				    PE03_MSCR_S32G PE03_PFE1_TX_EN_CFG
-				    PE04_MSCR_S32G PE04_PFE1_TX_D0_CFG
-				    PE05_MSCR_S32G PE05_PFE1_TX_D1_CFG
-				    PE06_MSCR_S32G PE06_PFE1_TX_D2_CFG
-				    PE07_MSCR_S32G PE07_PFE1_TX_D3_CFG
-				    PE08_MSCR_S32G PE08_PFE1_RX_CLK_CFG
-				    PE09_MSCR_S32G PE09_PFE1_RX_DV_CFG
-				    PE10_MSCR_S32G PE10_PFE1_RX_D0_CFG
-				    PE11_MSCR_S32G PE11_PFE1_RX_D1_CFG
-				    PE12_MSCR_S32G PE12_PFE1_RX_D2_CFG
-				    PE13_MSCR_S32G PE13_PFE1_RX_D3_CFG
-				    PFE1_RX_CLK_IMCR PE08_PFE1_RX_CLK_IN
-				    PFE1_RX_DV_IMCR PE09_PFE1_RX_DV_IN
-				    PFE1_RX_D0_IMCR PE10_PFE1_RX_D0_IN
-				    PFE1_RX_D1_IMCR PE11_PFE1_RX_D1_IN
-				    PFE1_RX_D2_IMCR PE12_PFE1_RX_D2_IN
-				    PFE1_RX_D3_IMCR PE13_PFE1_RX_D3_IN
-				    >;
+		pfe1_grp0 {
+			pinmux = <S32CC_PINMUX(66, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-up;
 		};
+
+		pfe1_grp1 {
+			pinmux = <S32CC_PINMUX(67, FUNC2)>,
+				 <S32CC_PINMUX(68, FUNC2)>,
+				 <S32CC_PINMUX(69, FUNC2)>,
+				 <S32CC_PINMUX(70, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		pfe1_grp2 {
+			pinmux = <S32CC_PINMUX(71, FUNC3)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		pfe1_grp3 {
+			pinmux = <S32CC_PINMUX(72, FUNC0)>,
+				 <S32CC_PINMUX(73, FUNC0)>,
+				 <S32CC_PINMUX(74, FUNC0)>,
+				 <S32CC_PINMUX(75, FUNC0)>,
+				 <S32CC_PINMUX(76, FUNC0)>,
+				 <S32CC_PINMUX(77, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		pfe1_grp4 {
+			pinmux = <S32CC_PINMUX(859, FUNC2)>,
+				 <S32CC_PINMUX(865, FUNC2)>,
+				 <S32CC_PINMUX(861, FUNC2)>,
+				 <S32CC_PINMUX(862, FUNC2)>,
+				 <S32CC_PINMUX(863, FUNC2)>,
+				 <S32CC_PINMUX(864, FUNC2)>;
+		};
+
 	};
 
 	pfe2_mdio_pins: pfe2_mdio {
-		pfe2_mdio_grp {
-			fsl,pins = <PE15_MSCR_S32G PE15_PFE2_MDIO_CFG
-				    PF02_MSCR_S32G PF02_PFE2_MDC_CFG
-				    PFE2_MDIO_IMCR PE15_PFE2_MDIO_IN
-				    >;
+		pfe2_mdio_grp0 {
+			pinmux = <S32CC_PINMUX(79, FUNC2)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		pfe2_mdio_grp1 {
+			pinmux = <S32CC_PINMUX(82, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
 		};
+
+		pfe2_mdio_grp2 {
+			pinmux = <S32CC_PINMUX(877, FUNC3)>;
+		};
+
 	};
 
 	pfe2_pins: pfe2 {
-		pfe2_grp {
-			fsl,pins = <PE14_MSCR_S32G PE14_PFE2_TX_EN_CFG
-				    PH01_MSCR_S32G PH01_PFE2_TX_D1_CFG
-				    PH02_MSCR_S32G PH02_PFE2_TX_D2_CFG
-				    PH03_MSCR_S32G PH03_PFE2_TX_D3_CFG
-				    PH04_MSCR_S32G PH04_PFE2_RX_CLK_CFG
-				    PH05_MSCR_S32G PH05_PFE2_RX_DV_CFG
-				    PH06_MSCR_S32G PH06_PFE2_RX_D0_CFG
-				    PH07_MSCR_S32G PH07_PFE2_RX_D1_CFG
-				    PH08_MSCR_S32G PH08_PFE2_RX_D2_CFG
-				    PH09_MSCR_S32G PH09_PFE2_RX_D3_CFG
-				    PH10_MSCR_S32G PH10_PFE2_TX_CLK_CFG
-				    PJ00_MSCR_S32G PJ00_PFE2_TX_D0_CFG
-				    PFE2_RX_CLK_IMCR PH04_PFE2_RX_CLK_IN
-				    PFE2_RX_DV_IMCR PH05_PFE2_RX_DV_IN
-				    PFE2_RX_D0_IMCR PH06_PFE2_RX_D0_IN
-				    PFE2_RX_D1_IMCR PH07_PFE2_RX_D1_IN
-				    PFE2_RX_D2_IMCR PH08_PFE2_RX_D2_IN
-				    PFE2_RX_D3_IMCR PH09_PFE2_RX_D3_IN
-				    >;
+		pfe2_grp0 {
+			pinmux = <S32CC_PINMUX(78, FUNC2)>,
+				 <S32CC_PINMUX(113, FUNC2)>,
+				 <S32CC_PINMUX(114, FUNC2)>,
+				 <S32CC_PINMUX(115, FUNC2)>,
+				 <S32CC_PINMUX(144, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		pfe2_grp1 {
+			pinmux = <S32CC_PINMUX(116, FUNC0)>,
+				 <S32CC_PINMUX(117, FUNC0)>,
+				 <S32CC_PINMUX(118, FUNC0)>,
+				 <S32CC_PINMUX(119, FUNC0)>,
+				 <S32CC_PINMUX(120, FUNC0)>,
+				 <S32CC_PINMUX(121, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		pfe2_grp2 {
+			pinmux = <S32CC_PINMUX(122, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-up;
+		};
+
+		pfe2_grp3 {
+			pinmux = <S32CC_PINMUX(879, FUNC3)>,
+				 <S32CC_PINMUX(885, FUNC3)>,
+				 <S32CC_PINMUX(881, FUNC3)>,
+				 <S32CC_PINMUX(882, FUNC3)>,
+				 <S32CC_PINMUX(883, FUNC3)>,
+				 <S32CC_PINMUX(884, FUNC3)>;
 		};
-	};
 	};
 };
 
diff --git a/fdts/s32g274a-bluebox3.dts b/fdts/s32g274a-bluebox3.dts
index 2d8bb20b9..665a2099d 100644
--- a/fdts/s32g274a-bluebox3.dts
+++ b/fdts/s32g274a-bluebox3.dts
@@ -140,150 +140,282 @@
 };
 
 &pinctrl {
-	board_pinctrl {
-		u-boot,dm-pre-reloc;
+	u-boot,dm-pre-reloc;
+
 	dspi0_pins: dspi0 {
-		dspi0_grp {
-			fsl,pins = <PA13_MSCR_S32G PA13_SPI0_SCK_CFG
-				    PA14_MSCR_S32G PA14_SPI0_SIN_CFG
-				    PA15_MSCR_S32G PA15_SPI0_SOUT_CFG
-				    PB09_MSCR_S32G PB09_SPI0_CS1_CFG
-				    PB10_MSCR_S32G PB10_SPI0_CS2_CFG
-				    DSPI0_SIN_IMCR PA14_SPI0_SIN_IN
-				    >;
+		dspi0_grp0 {
+			pinmux = <S32CC_PINMUX(13, FUNC1)>,
+				 <S32CC_PINMUX(15, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi0_grp1 {
+			pinmux = <S32CC_PINMUX(14, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi0_grp2 {
+			pinmux = <S32CC_PINMUX(25, FUNC5)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi0_grp3 {
+			pinmux = <S32CC_PINMUX(26, FUNC3)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
 		};
+
+		dspi0_grp4 {
+			pinmux = <S32CC_PINMUX(982, FUNC2)>;
+		};
+
 	};
 
 	dspi1_pins: dspi1 {
-		dspi1_grp {
-			fsl,pins = <PA06_MSCR_S32G PA06_SPI1_SOUT_CFG
-				    PA07_MSCR_S32G PA07_SPI1_CS0_CFG
-				    PA08_MSCR_S32G PA08_SPI1_SCK_CFG
-				    PE04_MSCR_S32G PE04_SPI1_CS3_CFG
-				    PF15_MSCR_S32G PF15_SPI1_SIN_CFG
-				    DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
-				    >;
+		dspi1_grp0 {
+			pinmux = <S32CC_PINMUX(6, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi1_grp1 {
+			pinmux = <S32CC_PINMUX(7, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
 		};
+
+		dspi1_grp2 {
+			pinmux = <S32CC_PINMUX(8, FUNC3)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi1_grp3 {
+			pinmux = <S32CC_PINMUX(68, FUNC3)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi1_grp4 {
+			pinmux = <S32CC_PINMUX(95, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi1_grp5 {
+			pinmux = <S32CC_PINMUX(987, FUNC2)>;
+		};
+
 	};
 
 	i2c0_pins: i2c0 {
-		i2c0_grp {
-			fsl,pins = <PB00_MSCR_S32G PB00_I2C0_SDA_CFG
-				    PB01_MSCR_S32G PB01_I2C0_SCL_CFG
-				    I2C0_SDA_IMCR PB00_I2C0_SDA_IN
-				    I2C0_SCL_IMCR PB01_I2C0_SCL_IN
-				    >;
+		i2c0_grp0 {
+			pinmux = <S32CC_PINMUX(16, FUNC1)>,
+				 <S32CC_PINMUX(17, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c0_grp1 {
+			pinmux = <S32CC_PINMUX(565, FUNC2)>,
+				 <S32CC_PINMUX(566, FUNC2)>;
 		};
+
 	};
 
 	i2c0_gpio_pins: i2c0_gpio {
-		i2c0_gpio_grp {
-			fsl,pins = <PB00_MSCR_S32G PB00_I2C0_SDA_GPIO
-				    PB01_MSCR_S32G PB01_I2C0_SCL_GPIO
-				    I2C0_SDA_IMCR IMCR_DISABLED
-				    I2C0_SCL_IMCR IMCR_DISABLED
-				    >;
+		i2c0_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(16, FUNC0)>,
+				 <S32CC_PINMUX(17, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		i2c0_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(565, FUNC0)>,
+				 <S32CC_PINMUX(566, FUNC0)>;
 		};
+
 	};
 
 	i2c1_pins: i2c1 {
-		i2c1_grp {
-			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_CFG
-				    PB04_MSCR_S32G PB04_I2C1_SDA_CFG
-				    I2C1_SCL_IMCR PB03_I2C1_SCL_IN
-				    I2C1_SDA_IMCR PB04_I2C1_SDA_IN
-				    >;
+		i2c1_grp0 {
+			pinmux = <S32CC_PINMUX(19, FUNC1)>,
+				 <S32CC_PINMUX(20, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c1_grp1 {
+			pinmux = <S32CC_PINMUX(717, FUNC2)>,
+				 <S32CC_PINMUX(718, FUNC2)>;
 		};
+
 	};
 
 	i2c1_gpio_pins: i2c1_gpio {
-		i2c1_gpio_grp {
-			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_GPIO
-				    PB04_MSCR_S32G PB04_I2C1_SDA_GPIO
-				    I2C1_SCL_IMCR IMCR_DISABLED
-				    I2C1_SDA_IMCR IMCR_DISABLED
-				    >;
+		i2c1_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(19, FUNC0)>,
+				 <S32CC_PINMUX(20, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		i2c1_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(717, FUNC0)>,
+				 <S32CC_PINMUX(718, FUNC0)>;
 		};
+
 	};
 
 	i2c2_pins: i2c2 {
-		i2c2_grp {
-			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_CFG
-				    PB06_MSCR_S32G PB06_I2C2_SDA_CFG
-				    I2C2_SCL_IMCR PB05_I2C2_SCL_IN
-				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
-				    >;
+		i2c2_grp0 {
+			pinmux = <S32CC_PINMUX(21, FUNC1)>,
+				 <S32CC_PINMUX(22, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
 		};
+
+		i2c2_grp1 {
+			pinmux = <S32CC_PINMUX(719, FUNC2)>,
+				 <S32CC_PINMUX(720, FUNC2)>;
+		};
+
 	};
 
 	i2c2_gpio_pins: i2c2_gpio {
-		i2c2_gpio_grp {
-			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_GPIO
-				    PB06_MSCR_S32G PB06_I2C2_SDA_GPIO
-				    I2C2_SCL_IMCR IMCR_DISABLED
-				    I2C2_SDA_IMCR IMCR_DISABLED
-				    >;
+		i2c2_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(21, FUNC0)>,
+				 <S32CC_PINMUX(22, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
 		};
+
+		i2c2_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(719, FUNC0)>,
+				 <S32CC_PINMUX(720, FUNC0)>;
+		};
+
 	};
 
 	i2c4_pins: i2c4 {
-		i2c4_grp {
-			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_CFG
-				    PC02_MSCR_S32G PC02_I2C4_SCL_CFG
-				    I2C4_SDA_IMCR PC01_I2C4_SDA_IN
-				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
-				    >;
+		i2c4_grp0 {
+			pinmux = <S32CC_PINMUX(33, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c4_grp1 {
+			pinmux = <S32CC_PINMUX(34, FUNC2)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c4_grp2 {
+			pinmux = <S32CC_PINMUX(724, FUNC3)>,
+				 <S32CC_PINMUX(723, FUNC3)>;
 		};
+
 	};
 
 	i2c4_gpio_pins: i2c4_gpio {
-		i2c4_gpio_grp {
-			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_GPIO
-				    PC02_MSCR_S32G PC02_I2C4_SCL_GPIO
-				    I2C4_SDA_IMCR IMCR_DISABLED
-				    I2C4_SCL_IMCR IMCR_DISABLED
-				    >;
+		i2c4_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(33, FUNC0)>,
+				 <S32CC_PINMUX(34, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		i2c4_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(724, FUNC0)>,
+				 <S32CC_PINMUX(723, FUNC0)>;
 		};
+
 	};
 
 	qspi_pins: qspi {
-		qspi_grp {
-			fsl,pins = <PF05_MSCR_S32G PF05_QSPI_DATA_A0_CFG
-				    PF06_MSCR_S32G PF06_QSPI_DATA_A1_CFG
-				    PF07_MSCR_S32G PF07_QSPI_DATA_A2_CFG
-				    PF08_MSCR_S32G PF08_QSPI_DATA_A3_CFG
-				    PF09_MSCR_S32G PF09_QSPI_DATA_A4_CFG
-				    PF10_MSCR_S32G PF10_QSPI_DATA_A5_CFG
-				    PF11_MSCR_S32G PF11_QSPI_DATA_A6_CFG
-				    PF12_MSCR_S32G PF12_QSPI_DATA_A7_CFG
-				    PF13_MSCR_S32G PF13_QSPI_DQS_A_CFG
-				    PG00_MSCR_S32G PG00_QSPI_CLK_A_CFG
-				    PG01_MSCR_S32G PG01_QSPI_CLK_A_b_CFG
-				    PG02_MSCR_S32G PG02_QSPI_CLK_2A_CFG
-				    PG03_MSCR_S32G PG03_QSPI_CLK_2A_b_CFG
-				    PG04_MSCR_S32G PG04_QSPI_CS_A0
-				    PG05_MSCR_S32G PG05_QSPI_CS_A1
-				    QSPI_DATA0_A_IMCR PF05_QSPI_DATA_A0_IN
-				    QSPI_DATA1_A_IMCR PF06_QSPI_DATA_A1_IN
-				    QSPI_DATA2_A_IMCR PF07_QSPI_DATA_A2_IN
-				    QSPI_DATA3_A_IMCR PF08_QSPI_DATA_A3_IN
-				    QSPI_DATA4_A_IMCR PF09_QSPI_DATA_A4_IN
-				    QSPI_DATA5_A_IMCR PF10_QSPI_DATA_A5_IN
-				    QSPI_DATA6_A_IMCR PF11_QSPI_DATA_A6_IN
-				    QSPI_DATA7_A_IMCR PF12_QSPI_DATA_A7_IN
-				    QSPI_DQS_A_IMCR PF13_DQS_A_IN
-				    >;
+		qspi_grp0 {
+			pinmux = <S32CC_PINMUX(85, FUNC1)>,
+				 <S32CC_PINMUX(86, FUNC1)>,
+				 <S32CC_PINMUX(87, FUNC1)>,
+				 <S32CC_PINMUX(88, FUNC1)>,
+				 <S32CC_PINMUX(89, FUNC1)>,
+				 <S32CC_PINMUX(90, FUNC1)>,
+				 <S32CC_PINMUX(91, FUNC1)>,
+				 <S32CC_PINMUX(92, FUNC1)>,
+				 <S32CC_PINMUX(93, FUNC1)>;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-down;
+		};
+
+		qspi_grp1 {
+			pinmux = <S32CC_PINMUX(96, FUNC1)>,
+				 <S32CC_PINMUX(97, FUNC1)>,
+				 <S32CC_PINMUX(98, FUNC1)>,
+				 <S32CC_PINMUX(99, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		qspi_grp2 {
+			pinmux = <S32CC_PINMUX(100, FUNC1)>,
+				 <S32CC_PINMUX(101, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-up;
+		};
+
+		qspi_grp3 {
+			pinmux = <S32CC_PINMUX(540, FUNC2)>,
+				 <S32CC_PINMUX(541, FUNC2)>,
+				 <S32CC_PINMUX(542, FUNC2)>,
+				 <S32CC_PINMUX(543, FUNC2)>,
+				 <S32CC_PINMUX(544, FUNC2)>,
+				 <S32CC_PINMUX(545, FUNC2)>,
+				 <S32CC_PINMUX(546, FUNC2)>,
+				 <S32CC_PINMUX(547, FUNC2)>,
+				 <S32CC_PINMUX(548, FUNC2)>;
 		};
+
 	};
 
 	uart0_pins: uart0 {
-		uart0_grp {
-			u-boot,dm-pre-reloc;
-			fsl,pins = <PC09_MSCR_S32G PC09_LIN0_TX_CFG
-				    PC10_MSCR_S32G PC10_LIN0_RX_CFG
-				    LIN0_RX_IMCR PC10_LIN0_RX_IN>;
+		uart0_grp0 {
+			pinmux = <S32CC_PINMUX(41, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		uart0_grp1 {
+			pinmux = <S32CC_PINMUX(42, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
 		};
-	};
 	};
 };
 
diff --git a/fdts/s32g274a-emu.dts b/fdts/s32g274a-emu.dts
index 9ed2ff729..5f44035a6 100644
--- a/fdts/s32g274a-emu.dts
+++ b/fdts/s32g274a-emu.dts
@@ -16,17 +16,24 @@
 };
 
 &pinctrl {
-	board_pinctrl {
-		u-boot,dm-pre-reloc;
+	u-boot,dm-pre-reloc;
+
 	uart1_pins: uart1 {
-		uart1_grp {
-			u-boot,dm-pre-reloc;
-			fsl,pins = <PB09_MSCR_S32G PB09_LIN1_TX_CFG
-				    PB10_MSCR_S32G PB10_LIN1_RX_CFG
-				    LIN1_RX_IMCR PB10_LIN1_RX_IN
-				    >;
+		uart1_grp0 {
+			pinmux = <S32CC_PINMUX(25, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		uart1_grp1 {
+			pinmux = <S32CC_PINMUX(26, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		uart1_grp2 {
+			pinmux = <S32CC_PINMUX(736, FUNC3)>;
 		};
-	};
 	};
 };
 
diff --git a/fdts/s32g399a-emu.dts b/fdts/s32g399a-emu.dts
index 4da3ad8a1..7653790d2 100644
--- a/fdts/s32g399a-emu.dts
+++ b/fdts/s32g399a-emu.dts
@@ -17,17 +17,24 @@
 };
 
 &pinctrl {
-	board_pinctrl {
-		u-boot,dm-pre-reloc;
+	u-boot,dm-pre-reloc;
+
 	uart1_pins: uart1 {
-		uart1_grp {
-			u-boot,dm-pre-reloc;
-			fsl,pins = <PB09_MSCR_S32G PB09_LIN1_TX_CFG
-				    PB10_MSCR_S32G PB10_LIN1_RX_CFG
-				    LIN1_RX_IMCR PB10_LIN1_RX_IN
-				    >;
+		uart1_grp0 {
+			pinmux = <S32CC_PINMUX(25, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		uart1_grp1 {
+			pinmux = <S32CC_PINMUX(26, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		uart1_grp2 {
+			pinmux = <S32CC_PINMUX(736, FUNC3)>;
 		};
-	};
 	};
 };
 
diff --git a/fdts/s32gxxxa-evb.dtsi b/fdts/s32gxxxa-evb.dtsi
index 75b54ec6f..c3d9bc4cb 100644
--- a/fdts/s32gxxxa-evb.dtsi
+++ b/fdts/s32gxxxa-evb.dtsi
@@ -148,177 +148,313 @@
 };
 
 &pinctrl {
-	board_pinctrl {
-		u-boot,dm-pre-reloc;
+	u-boot,dm-pre-reloc;
+
 	i2c0_pins: i2c0 {
-		i2c0_grp {
-			fsl,pins = <PB00_MSCR_S32G PB00_I2C0_SDA_CFG
-				    PB01_MSCR_S32G PB01_I2C0_SCL_CFG
-				    I2C0_SDA_IMCR PB00_I2C0_SDA_IN
-				    I2C0_SCL_IMCR PB01_I2C0_SCL_IN
-				    >;
+		i2c0_grp0 {
+			pinmux = <S32CC_PINMUX(16, FUNC1)>,
+				 <S32CC_PINMUX(17, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c0_grp1 {
+			pinmux = <S32CC_PINMUX(565, FUNC2)>,
+				 <S32CC_PINMUX(566, FUNC2)>;
 		};
+
 	};
 
 	i2c0_gpio_pins: i2c0_gpio {
-		i2c0_gpio_grp {
-			fsl,pins = <PB00_MSCR_S32G PB00_I2C0_SDA_GPIO
-				    PB01_MSCR_S32G PB01_I2C0_SCL_GPIO
-				    I2C0_SDA_IMCR IMCR_DISABLED
-				    I2C0_SCL_IMCR IMCR_DISABLED
-				    >;
+		i2c0_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(16, FUNC0)>,
+				 <S32CC_PINMUX(17, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
 		};
+
+		i2c0_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(565, FUNC0)>,
+				 <S32CC_PINMUX(566, FUNC0)>;
+		};
+
 	};
 
 	i2c1_pins: i2c1 {
-		i2c1_grp {
-			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_CFG
-				    PB04_MSCR_S32G PB04_I2C1_SDA_CFG
-				    I2C1_SCL_IMCR PB03_I2C1_SCL_IN
-				    I2C1_SDA_IMCR PB04_I2C1_SDA_IN
-				    >;
+		i2c1_grp0 {
+			pinmux = <S32CC_PINMUX(19, FUNC1)>,
+				 <S32CC_PINMUX(20, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
 		};
+
+		i2c1_grp1 {
+			pinmux = <S32CC_PINMUX(717, FUNC2)>,
+				 <S32CC_PINMUX(718, FUNC2)>;
+		};
+
 	};
 
 	i2c1_gpio_pins: i2c1_gpio {
-		i2c1_gpio_grp {
-			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_GPIO
-				    PB04_MSCR_S32G PB04_I2C1_SDA_GPIO
-				    I2C1_SCL_IMCR IMCR_DISABLED
-				    I2C1_SDA_IMCR IMCR_DISABLED
-				    >;
+		i2c1_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(19, FUNC0)>,
+				 <S32CC_PINMUX(20, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		i2c1_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(717, FUNC0)>,
+				 <S32CC_PINMUX(718, FUNC0)>;
 		};
+
 	};
 
 	i2c2_pins: i2c2 {
-		i2c2_grp {
-			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_CFG
-				    PB06_MSCR_S32G PB06_I2C2_SDA_CFG
-				    I2C2_SCL_IMCR PB05_I2C2_SCL_IN
-				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
-				    >;
+		i2c2_grp0 {
+			pinmux = <S32CC_PINMUX(21, FUNC1)>,
+				 <S32CC_PINMUX(22, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c2_grp1 {
+			pinmux = <S32CC_PINMUX(719, FUNC2)>,
+				 <S32CC_PINMUX(720, FUNC2)>;
 		};
+
 	};
 
 	i2c2_gpio_pins: i2c2_gpio {
-		i2c2_gpio_grp {
-			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_GPIO
-				    PB06_MSCR_S32G PB06_I2C2_SDA_GPIO
-				    I2C2_SCL_IMCR IMCR_DISABLED
-				    I2C2_SDA_IMCR IMCR_DISABLED
-				    >;
+		i2c2_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(21, FUNC0)>,
+				 <S32CC_PINMUX(22, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		i2c2_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(719, FUNC0)>,
+				 <S32CC_PINMUX(720, FUNC0)>;
 		};
+
 	};
 
 	i2c4_pins: i2c4 {
-		i2c4_grp {
-			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_CFG
-				    PC02_MSCR_S32G PC02_I2C4_SCL_CFG
-				    I2C4_SDA_IMCR PC01_I2C4_SDA_IN
-				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
-				    >;
+		i2c4_grp0 {
+			pinmux = <S32CC_PINMUX(33, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c4_grp1 {
+			pinmux = <S32CC_PINMUX(34, FUNC2)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c4_grp2 {
+			pinmux = <S32CC_PINMUX(724, FUNC3)>,
+				 <S32CC_PINMUX(723, FUNC3)>;
 		};
+
 	};
 
 	i2c4_gpio_pins: i2c4_gpio {
-		i2c4_gpio_grp {
-			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_GPIO
-				    PC02_MSCR_S32G PC02_I2C4_SCL_GPIO
-				    I2C4_SDA_IMCR IMCR_DISABLED
-				    I2C4_SCL_IMCR IMCR_DISABLED
-				    >;
+		i2c4_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(33, FUNC0)>,
+				 <S32CC_PINMUX(34, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		i2c4_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(724, FUNC0)>,
+				 <S32CC_PINMUX(723, FUNC0)>;
 		};
+
 	};
 
 	qspi_pins: qspi {
-		qspi_grp {
-			fsl,pins = <PF05_MSCR_S32G PF05_QSPI_DATA_A0_CFG
-				    PF06_MSCR_S32G PF06_QSPI_DATA_A1_CFG
-				    PF07_MSCR_S32G PF07_QSPI_DATA_A2_CFG
-				    PF08_MSCR_S32G PF08_QSPI_DATA_A3_CFG
-				    PF09_MSCR_S32G PF09_QSPI_DATA_A4_CFG
-				    PF10_MSCR_S32G PF10_QSPI_DATA_A5_CFG
-				    PF11_MSCR_S32G PF11_QSPI_DATA_A6_CFG
-				    PF12_MSCR_S32G PF12_QSPI_DATA_A7_CFG
-				    PF13_MSCR_S32G PF13_QSPI_DQS_A_CFG
-				    PG00_MSCR_S32G PG00_QSPI_CLK_A_CFG
-				    PG01_MSCR_S32G PG01_QSPI_CLK_A_b_CFG
-				    PG02_MSCR_S32G PG02_QSPI_CLK_2A_CFG
-				    PG03_MSCR_S32G PG03_QSPI_CLK_2A_b_CFG
-				    PG04_MSCR_S32G PG04_QSPI_CS_A0
-				    PG05_MSCR_S32G PG05_QSPI_CS_A1
-				    QSPI_DATA0_A_IMCR PF05_QSPI_DATA_A0_IN
-				    QSPI_DATA1_A_IMCR PF06_QSPI_DATA_A1_IN
-				    QSPI_DATA2_A_IMCR PF07_QSPI_DATA_A2_IN
-				    QSPI_DATA3_A_IMCR PF08_QSPI_DATA_A3_IN
-				    QSPI_DATA4_A_IMCR PF09_QSPI_DATA_A4_IN
-				    QSPI_DATA5_A_IMCR PF10_QSPI_DATA_A5_IN
-				    QSPI_DATA6_A_IMCR PF11_QSPI_DATA_A6_IN
-				    QSPI_DATA7_A_IMCR PF12_QSPI_DATA_A7_IN
-				    QSPI_DQS_A_IMCR PF13_DQS_A_IN
-				    >;
+		qspi_grp0 {
+			pinmux = <S32CC_PINMUX(85, FUNC1)>,
+				 <S32CC_PINMUX(86, FUNC1)>,
+				 <S32CC_PINMUX(87, FUNC1)>,
+				 <S32CC_PINMUX(88, FUNC1)>,
+				 <S32CC_PINMUX(89, FUNC1)>,
+				 <S32CC_PINMUX(90, FUNC1)>,
+				 <S32CC_PINMUX(91, FUNC1)>,
+				 <S32CC_PINMUX(92, FUNC1)>,
+				 <S32CC_PINMUX(93, FUNC1)>;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-down;
+		};
+
+		qspi_grp1 {
+			pinmux = <S32CC_PINMUX(96, FUNC1)>,
+				 <S32CC_PINMUX(97, FUNC1)>,
+				 <S32CC_PINMUX(98, FUNC1)>,
+				 <S32CC_PINMUX(99, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		qspi_grp2 {
+			pinmux = <S32CC_PINMUX(100, FUNC1)>,
+				 <S32CC_PINMUX(101, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-up;
+		};
+
+		qspi_grp3 {
+			pinmux = <S32CC_PINMUX(540, FUNC2)>,
+				 <S32CC_PINMUX(541, FUNC2)>,
+				 <S32CC_PINMUX(542, FUNC2)>,
+				 <S32CC_PINMUX(543, FUNC2)>,
+				 <S32CC_PINMUX(544, FUNC2)>,
+				 <S32CC_PINMUX(545, FUNC2)>,
+				 <S32CC_PINMUX(546, FUNC2)>,
+				 <S32CC_PINMUX(547, FUNC2)>,
+				 <S32CC_PINMUX(548, FUNC2)>;
 		};
+
 	};
 
 	dspi1_pins: dspi1 {
-		dspi1_grp {
-			fsl,pins = <PA06_MSCR_S32G PA06_SPI1_SOUT_CFG
-				    PA07_MSCR_S32G PA07_SPI1_CS0_CFG
-				    PA08_MSCR_S32G PA08_SPI1_SCK_CFG
-				    PF15_MSCR_S32G PF15_SPI1_SIN_CFG
-				    DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
-				    >;
+		dspi1_grp0 {
+			pinmux = <S32CC_PINMUX(6, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi1_grp1 {
+			pinmux = <S32CC_PINMUX(7, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi1_grp2 {
+			pinmux = <S32CC_PINMUX(8, FUNC3)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
 		};
+
+		dspi1_grp3 {
+			pinmux = <S32CC_PINMUX(95, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi1_grp4 {
+			pinmux = <S32CC_PINMUX(987, FUNC2)>;
+		};
+
 	};
 
 	dspi5_pins: dspi5 {
-		dspi5_grp {
-			fsl,pins = <PA09_MSCR_S32G PA09_SPI5_SCK_CFG
-				    PA10_MSCR_S32G PA10_SPI5_SIN_CFG
-				    PA11_MSCR_S32G PA11_SPI5_SOUT_CFG
-				    PA12_MSCR_S32G PA12_SPI5_CS0_CFG
-				    DSPI5_SIN_IMCR PA10_SPI5_SIN_IN
-				    >;
+		dspi5_grp0 {
+			pinmux = <S32CC_PINMUX(9, FUNC3)>,
+				 <S32CC_PINMUX(11, FUNC3)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi5_grp1 {
+			pinmux = <S32CC_PINMUX(10, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
 		};
+
+		dspi5_grp2 {
+			pinmux = <S32CC_PINMUX(12, FUNC3)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi5_grp3 {
+			pinmux = <S32CC_PINMUX(1007, FUNC2)>;
+		};
+
 	};
 
 	usb_pins: usb {
-		usb_grp {
-			fsl,pins = <PD14_MSCR_S32G PD14_USB_DATA0_CFG
-				    PD15_MSCR_S32G PD15_USB_DATA1_CFG
-				    PE00_MSCR_S32G PE00_USB_DATA2_CFG
-				    PE01_MSCR_S32G PE01_USB_DATA3_CFG
-				    PH00_MSCR_S32G PH00_USB_DATA7_CFG
-				    PL08_MSCR_S32G PL08_USB_CLK_CFG
-				    PL09_MSCR_S32G PL09_USB_DIR_CFG
-				    PL10_MSCR_S32G PL10_USB_STP_CFG
-				    PL11_MSCR_S32G PL11_USB_NXT_CFG
-				    PL12_MSCR_S32G PL12_USB_DATA4_CFG
-				    PL13_MSCR_S32G PL13_USB_DATA5_CFG
-				    PL14_MSCR_S32G PL14_USB_DATA6_CFG
-				    USB_ULPI_CLK_IMCR PL08_USB_CLK_IN
-				    USB_ULPI_DATA0_IMCR PD14_USB_DATA0_IN
-				    USB_ULPI_DATA1_IMCR PD15_USB_DATA1_IN
-				    USB_ULPI_DATA2_IMCR PE00_USB_DATA2_IN
-				    USB_ULPI_DATA3_IMCR PE01_USB_DATA3_IN
-				    USB_ULPI_DATA4_IMCR PL12_USB_DATA4_IN
-				    USB_ULPI_DATA5_IMCR PL13_USB_DATA5_IN
-				    USB_ULPI_DATA6_IMCR PL14_USB_DATA6_IN
-				    USB_ULPI_DATA7_IMCR PH00_USB_DATA7_IN
-				    USB_ULPI_DIR_IMCR PL09_USB_DIR_IN
-				    USB_ULPI_NXT_IMCR PL11_USB_NXT_IN
-				    >;
+		usb_grp0 {
+			pinmux = <S32CC_PINMUX(62, FUNC1)>,
+				 <S32CC_PINMUX(63, FUNC1)>,
+				 <S32CC_PINMUX(64, FUNC1)>,
+				 <S32CC_PINMUX(65, FUNC1)>,
+				 <S32CC_PINMUX(112, FUNC1)>,
+				 <S32CC_PINMUX(188, FUNC1)>,
+				 <S32CC_PINMUX(189, FUNC1)>,
+				 <S32CC_PINMUX(190, FUNC1)>;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		usb_grp1 {
+			pinmux = <S32CC_PINMUX(184, FUNC0)>,
+				 <S32CC_PINMUX(185, FUNC0)>,
+				 <S32CC_PINMUX(187, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
 		};
+
+		usb_grp2 {
+			pinmux = <S32CC_PINMUX(186, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		usb_grp3 {
+			pinmux = <S32CC_PINMUX(895, FUNC2)>,
+				 <S32CC_PINMUX(896, FUNC2)>,
+				 <S32CC_PINMUX(897, FUNC2)>,
+				 <S32CC_PINMUX(898, FUNC2)>,
+				 <S32CC_PINMUX(899, FUNC2)>,
+				 <S32CC_PINMUX(900, FUNC2)>,
+				 <S32CC_PINMUX(901, FUNC2)>,
+				 <S32CC_PINMUX(902, FUNC2)>,
+				 <S32CC_PINMUX(903, FUNC2)>,
+				 <S32CC_PINMUX(904, FUNC2)>,
+				 <S32CC_PINMUX(905, FUNC2)>;
+		};
+
 	};
 
 	uart0_pins: uart0 {
-		uart0_grp {
-			u-boot,dm-pre-reloc;
-			fsl,pins = <PC09_MSCR_S32G PC09_LIN0_TX_CFG
-				    PC10_MSCR_S32G PC10_LIN0_RX_CFG
-				    LIN0_RX_IMCR PC10_LIN0_RX_IN>;
+		uart0_grp0 {
+			pinmux = <S32CC_PINMUX(41, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		uart0_grp1 {
+			pinmux = <S32CC_PINMUX(42, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
 		};
-	};
 	};
 };
 
diff --git a/fdts/s32gxxxa-rdb.dtsi b/fdts/s32gxxxa-rdb.dtsi
index 983868979..267bc04fa 100644
--- a/fdts/s32gxxxa-rdb.dtsi
+++ b/fdts/s32gxxxa-rdb.dtsi
@@ -128,148 +128,278 @@
 };
 
 &pinctrl {
-	board_pinctrl {
-		u-boot,dm-pre-reloc;
+	u-boot,dm-pre-reloc;
+
 	i2c0_pins: i2c0 {
-		i2c0_grp {
-			fsl,pins = <PB15_MSCR_S32G PB15_I2C0_SDA_CFG
-				    PC00_MSCR_S32G PC00_I2C0_SCL_CFG
-				    I2C0_SDA_IMCR PB15_I2C0_SDA_IN
-				    I2C0_SCL_IMCR PC00_I2C0_SCL_IN
-				    >;
+		i2c0_grp0 {
+			pinmux = <S32CC_PINMUX(31, FUNC2)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c0_grp1 {
+			pinmux = <S32CC_PINMUX(32, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c0_grp2 {
+			pinmux = <S32CC_PINMUX(565, FUNC3)>,
+				 <S32CC_PINMUX(566, FUNC3)>;
 		};
+
 	};
 
 	i2c0_gpio_pins: i2c0_gpio {
-		i2c0_gpio_grp {
-			fsl,pins = <PB15_MSCR_S32G PB15_I2C0_SDA_GPIO
-				    PC00_MSCR_S32G PC00_I2C0_SCL_GPIO
-				    I2C0_SDA_IMCR IMCR_DISABLED
-				    I2C0_SCL_IMCR IMCR_DISABLED
-				    >;
+		i2c0_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(31, FUNC0)>,
+				 <S32CC_PINMUX(32, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		i2c0_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(565, FUNC0)>,
+				 <S32CC_PINMUX(566, FUNC0)>;
 		};
+
 	};
 
 	i2c1_pins: i2c1 {
-		i2c1_grp {
-			fsl,pins = <PK03_MSCR_S32G PK03_I2C1_SCL_CFG
-				    PK05_MSCR_S32G PK05_I2C1_SDA_CFG
-				    I2C1_SCL_IMCR PK03_I2C1_SCL_IN
-				    I2C1_SDA_IMCR PK05_I2C1_SDA_IN
-				    >;
+		i2c1_grp0 {
+			pinmux = <S32CC_PINMUX(163, FUNC3)>,
+				 <S32CC_PINMUX(165, FUNC3)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c1_grp1 {
+			pinmux = <S32CC_PINMUX(717, FUNC5)>;
 		};
+
+		i2c1_grp2 {
+			pinmux = <S32CC_PINMUX(718, FUNC4)>;
+		};
+
 	};
 
 	i2c1_gpio_pins: i2c1_gpio {
-		i2c1_gpio_grp {
-			fsl,pins = <PK03_MSCR_S32G PK03_I2C1_SCL_GPIO
-				    PK05_MSCR_S32G PK05_I2C1_SDA_GPIO
-				    I2C1_SCL_IMCR IMCR_DISABLED
-				    I2C1_SDA_IMCR IMCR_DISABLED
-				    >;
+		i2c1_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(163, FUNC0)>,
+				 <S32CC_PINMUX(165, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
 		};
+
+		i2c1_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(717, FUNC0)>,
+				 <S32CC_PINMUX(718, FUNC0)>;
+		};
+
 	};
 
 	i2c2_pins: i2c2 {
-		i2c2_grp {
-			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_CFG
-				    PB06_MSCR_S32G PB06_I2C2_SDA_CFG
-				    I2C2_SCL_IMCR PB05_I2C2_SCL_IN
-				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
-				    >;
+		i2c2_grp0 {
+			pinmux = <S32CC_PINMUX(21, FUNC1)>,
+				 <S32CC_PINMUX(22, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c2_grp1 {
+			pinmux = <S32CC_PINMUX(719, FUNC2)>,
+				 <S32CC_PINMUX(720, FUNC2)>;
 		};
+
 	};
 
 	i2c2_gpio_pins: i2c2_gpio {
-		i2c2_gpio_grp {
-			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_GPIO
-				    PB06_MSCR_S32G PB06_I2C2_SDA_GPIO
-				    I2C2_SCL_IMCR IMCR_DISABLED
-				    I2C2_SDA_IMCR IMCR_DISABLED
-				    >;
+		i2c2_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(21, FUNC0)>,
+				 <S32CC_PINMUX(22, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		i2c2_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(719, FUNC0)>,
+				 <S32CC_PINMUX(720, FUNC0)>;
 		};
+
 	};
 
 	i2c4_pins: i2c4 {
-		i2c4_grp {
-			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_CFG
-				    PC02_MSCR_S32G PC02_I2C4_SCL_CFG
-				    I2C4_SDA_IMCR PC01_I2C4_SDA_IN
-				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
-				    >;
+		i2c4_grp0 {
+			pinmux = <S32CC_PINMUX(33, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c4_grp1 {
+			pinmux = <S32CC_PINMUX(34, FUNC2)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
 		};
+
+		i2c4_grp2 {
+			pinmux = <S32CC_PINMUX(724, FUNC3)>,
+				 <S32CC_PINMUX(723, FUNC3)>;
+		};
+
 	};
 
 	i2c4_gpio_pins: i2c4_gpio {
-		i2c4_gpio_grp {
-			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_GPIO
-				    PC02_MSCR_S32G PC02_I2C4_SCL_GPIO
-				    I2C4_SDA_IMCR IMCR_DISABLED
-				    I2C4_SCL_IMCR IMCR_DISABLED
-				    >;
+		i2c4_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(33, FUNC0)>,
+				 <S32CC_PINMUX(34, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
 		};
+
+		i2c4_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(724, FUNC0)>,
+				 <S32CC_PINMUX(723, FUNC0)>;
+		};
+
 	};
 
 	qspi_pins: qspi {
-		qspi_grp {
-			fsl,pins = <PF05_MSCR_S32G PF05_QSPI_DATA_A0_CFG
-				    PF06_MSCR_S32G PF06_QSPI_DATA_A1_CFG
-				    PF07_MSCR_S32G PF07_QSPI_DATA_A2_CFG
-				    PF08_MSCR_S32G PF08_QSPI_DATA_A3_CFG
-				    PF09_MSCR_S32G PF09_QSPI_DATA_A4_CFG
-				    PF10_MSCR_S32G PF10_QSPI_DATA_A5_CFG
-				    PF11_MSCR_S32G PF11_QSPI_DATA_A6_CFG
-				    PF12_MSCR_S32G PF12_QSPI_DATA_A7_CFG
-				    PF13_MSCR_S32G PF13_QSPI_DQS_A_CFG
-				    PG00_MSCR_S32G PG00_QSPI_CLK_A_CFG
-				    PG01_MSCR_S32G PG01_QSPI_CLK_A_b_CFG
-				    PG02_MSCR_S32G PG02_QSPI_CLK_2A_CFG
-				    PG03_MSCR_S32G PG03_QSPI_CLK_2A_b_CFG
-				    PG04_MSCR_S32G PG04_QSPI_CS_A0
-				    PG05_MSCR_S32G PG05_QSPI_CS_A1
-				    QSPI_DATA0_A_IMCR PF05_QSPI_DATA_A0_IN
-				    QSPI_DATA1_A_IMCR PF06_QSPI_DATA_A1_IN
-				    QSPI_DATA2_A_IMCR PF07_QSPI_DATA_A2_IN
-				    QSPI_DATA3_A_IMCR PF08_QSPI_DATA_A3_IN
-				    QSPI_DATA4_A_IMCR PF09_QSPI_DATA_A4_IN
-				    QSPI_DATA5_A_IMCR PF10_QSPI_DATA_A5_IN
-				    QSPI_DATA6_A_IMCR PF11_QSPI_DATA_A6_IN
-				    QSPI_DATA7_A_IMCR PF12_QSPI_DATA_A7_IN
-				    QSPI_DQS_A_IMCR PF13_DQS_A_IN
-				    >;
+		qspi_grp0 {
+			pinmux = <S32CC_PINMUX(85, FUNC1)>,
+				 <S32CC_PINMUX(86, FUNC1)>,
+				 <S32CC_PINMUX(87, FUNC1)>,
+				 <S32CC_PINMUX(88, FUNC1)>,
+				 <S32CC_PINMUX(89, FUNC1)>,
+				 <S32CC_PINMUX(90, FUNC1)>,
+				 <S32CC_PINMUX(91, FUNC1)>,
+				 <S32CC_PINMUX(92, FUNC1)>,
+				 <S32CC_PINMUX(93, FUNC1)>;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-down;
+		};
+
+		qspi_grp1 {
+			pinmux = <S32CC_PINMUX(96, FUNC1)>,
+				 <S32CC_PINMUX(97, FUNC1)>,
+				 <S32CC_PINMUX(98, FUNC1)>,
+				 <S32CC_PINMUX(99, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		qspi_grp2 {
+			pinmux = <S32CC_PINMUX(100, FUNC1)>,
+				 <S32CC_PINMUX(101, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-up;
+		};
+
+		qspi_grp3 {
+			pinmux = <S32CC_PINMUX(540, FUNC2)>,
+				 <S32CC_PINMUX(541, FUNC2)>,
+				 <S32CC_PINMUX(542, FUNC2)>,
+				 <S32CC_PINMUX(543, FUNC2)>,
+				 <S32CC_PINMUX(544, FUNC2)>,
+				 <S32CC_PINMUX(545, FUNC2)>,
+				 <S32CC_PINMUX(546, FUNC2)>,
+				 <S32CC_PINMUX(547, FUNC2)>,
+				 <S32CC_PINMUX(548, FUNC2)>;
 		};
+
 	};
 
 	dspi1_pins: dspi1 {
-		dspi1_grp {
-			fsl,pins = <PA06_MSCR_S32G PA06_SPI1_SOUT_CFG
-				    PA07_MSCR_S32G PA07_SPI1_CS0_CFG
-				    PA08_MSCR_S32G PA08_SPI1_SCK_CFG
-				    PF15_MSCR_S32G PF15_SPI1_SIN_CFG
-				    DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
-				    >;
+		dspi1_grp0 {
+			pinmux = <S32CC_PINMUX(6, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi1_grp1 {
+			pinmux = <S32CC_PINMUX(7, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi1_grp2 {
+			pinmux = <S32CC_PINMUX(8, FUNC3)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi1_grp3 {
+			pinmux = <S32CC_PINMUX(95, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi1_grp4 {
+			pinmux = <S32CC_PINMUX(987, FUNC2)>;
 		};
+
 	};
 
 	dspi5_pins: dspi5 {
-		dspi5_grp {
-			fsl,pins = <PA09_MSCR_S32G PA09_SPI5_SCK_CFG
-				    PA10_MSCR_S32G PA10_SPI5_SIN_CFG
-				    PA11_MSCR_S32G PA11_SPI5_SOUT_CFG
-				    PA12_MSCR_S32G PA12_SPI5_CS0_CFG
-				    DSPI5_SIN_IMCR PA10_SPI5_SIN_IN
-				    >;
+		dspi5_grp0 {
+			pinmux = <S32CC_PINMUX(9, FUNC3)>,
+				 <S32CC_PINMUX(11, FUNC3)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi5_grp1 {
+			pinmux = <S32CC_PINMUX(10, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi5_grp2 {
+			pinmux = <S32CC_PINMUX(12, FUNC3)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi5_grp3 {
+			pinmux = <S32CC_PINMUX(1007, FUNC2)>;
 		};
+
 	};
 
 	uart0_pins: uart0 {
-		uart0_grp {
-			u-boot,dm-pre-reloc;
-			fsl,pins = <PC09_MSCR_S32G PC09_LIN0_TX_CFG
-				    PC10_MSCR_S32G PC10_LIN0_RX_CFG
-				    LIN0_RX_IMCR PC10_LIN0_RX_IN>;
+		uart0_grp0 {
+			pinmux = <S32CC_PINMUX(41, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		uart0_grp1 {
+			pinmux = <S32CC_PINMUX(42, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
 		};
-	};
 	};
 };
 
diff --git a/fdts/s32r45-evb.dts b/fdts/s32r45-evb.dts
index 81b4a39e2..b8cd118ef 100644
--- a/fdts/s32r45-evb.dts
+++ b/fdts/s32r45-evb.dts
@@ -88,194 +88,368 @@
 };
 
 &pinctrl {
-	board_pinctrl {
-		u-boot,dm-pre-reloc;
+	u-boot,dm-pre-reloc;
+
 	dspi1_pins: dspi1 {
-		dspi1_grp {
-			fsl,pins = <PB04_MSCR_S32R45 PB04_SPI1_SCK_CFG
-				    PB05_MSCR_S32R45 PB05_SPI1_SIN_CFG
-				    PB06_MSCR_S32R45 PB06_SPI1_SOUT_CFG
-				    PB07_MSCR_S32R45 PB07_SPI1_CS0_CFG
-				    PB08_MSCR_S32R45 PB08_SPI1_CS1_CFG
-				    DSPI1_SIN_IMCR PB05_SPI1_SIN_IN
-				    >;
+		dspi1_grp0 {
+			pinmux = <S32CC_PINMUX(20, FUNC1)>,
+				 <S32CC_PINMUX(22, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi1_grp1 {
+			pinmux = <S32CC_PINMUX(21, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi1_grp2 {
+			pinmux = <S32CC_PINMUX(23, FUNC1)>,
+				 <S32CC_PINMUX(24, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
 		};
+
+		dspi1_grp3 {
+			pinmux = <S32CC_PINMUX(623, FUNC4)>;
+		};
+
 	};
 
 	dspi2_pins: dspi2 {
-		dspi2_grp {
-			fsl,pins = <PB11_MSCR_S32R45 PB11_SPI2_SCK_CFG
-				    PB13_MSCR_S32R45 PB13_SPI2_SOUT_CFG
-				    PB14_MSCR_S32R45 PB14_SPI2_CS0_CFG
-				    PB15_MSCR_S32R45 PB15_SPI2_CS1_CFG
-				    PC01_MSCR_S32R45 PC01_SPI2_SIN_CFG
-				    DSPI2_SIN_IMCR PC01_SPI2_SIN_IN
-				    >;
+		dspi2_grp0 {
+			pinmux = <S32CC_PINMUX(27, FUNC1)>,
+				 <S32CC_PINMUX(29, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi2_grp1 {
+			pinmux = <S32CC_PINMUX(30, FUNC1)>,
+				 <S32CC_PINMUX(31, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
 		};
+
+		dspi2_grp2 {
+			pinmux = <S32CC_PINMUX(33, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi2_grp3 {
+			pinmux = <S32CC_PINMUX(681, FUNC2)>;
+		};
+
 	};
 
 	dspi3_pins: dspi3 {
-		dspi3_grp {
-			fsl,pins = <PC04_MSCR_S32R45 PC04_SPI3_SCK_CFG
-				    PC06_MSCR_S32R45 PC06_SPI3_SIN_CFG
-				    PC07_MSCR_S32R45 PC07_SPI3_CS0_CFG
-				    PC08_MSCR_S32R45 PC08_SPI3_CS1_CFG
-				    PC13_MSCR_S32R45 PC13_SPI3_SOUT_CFG
-				    DSPI3_SIN_IMCR PC06_SPI3_SIN_IN
-				    >;
+		dspi3_grp0 {
+			pinmux = <S32CC_PINMUX(36, FUNC1)>,
+				 <S32CC_PINMUX(45, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi3_grp1 {
+			pinmux = <S32CC_PINMUX(38, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi3_grp2 {
+			pinmux = <S32CC_PINMUX(39, FUNC1)>,
+				 <S32CC_PINMUX(40, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
 		};
+
+		dspi3_grp3 {
+			pinmux = <S32CC_PINMUX(645, FUNC2)>;
+		};
+
 	};
 
 	dspi5_pins: dspi5 {
-		dspi5_grp {
-			fsl,pins = <PK00_MSCR_S32R45 PK00_SPI5_SCK_CFG
-				    PK03_MSCR_S32R45 PK03_SPI5_CS0_CFG
-				    PK04_MSCR_S32R45 PK04_SPI5_SIN_CFG
-				    PK05_MSCR_S32R45 PK05_SPI5_SOUT_CFG
-				    DSPI5_SIN_IMCR PK04_SPI5_SIN_IN
-				    >;
+		dspi5_grp0 {
+			pinmux = <S32CC_PINMUX(128, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		dspi5_grp1 {
+			pinmux = <S32CC_PINMUX(131, FUNC2)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi5_grp2 {
+			pinmux = <S32CC_PINMUX(132, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+			bias-pull-up;
+		};
+
+		dspi5_grp3 {
+			pinmux = <S32CC_PINMUX(133, FUNC4)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
 		};
+
+		dspi5_grp4 {
+			pinmux = <S32CC_PINMUX(750, FUNC4)>;
+		};
+
 	};
 
 	gmac0_mdio_pins: gmac0_mdio {
-		gmac0_mdio_grp {
-			fsl,pins = <PD12_MSCR_S32R45 PD12_GMAC0_MDC_CFG
-				    PD13_MSCR_S32R45 PD13_GMAC0_MDIO_CFG
-				    GMAC0_MDIO_IMCR PD13_GMAC0_MDIO_IN
-				    >;
+		gmac0_mdio_grp0 {
+			pinmux = <S32CC_PINMUX(60, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		gmac0_mdio_grp1 {
+			pinmux = <S32CC_PINMUX(61, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
 		};
+
+		gmac0_mdio_grp2 {
+			pinmux = <S32CC_PINMUX(527, FUNC2)>;
+		};
+
 	};
 
 	gmac0_pins: gmac0 {
-		gmac0_grp {
-			fsl,pins = <PE02_MSCR_S32R45 PE02_GMAC0_TX_CLK_CFG
-				    PE03_MSCR_S32R45 PE03_GMAC0_TX_EN_CFG
-				    PE04_MSCR_S32R45 PE04_GMAC0_TX_D0_CFG
-				    PE05_MSCR_S32R45 PE05_GMAC0_TX_D1_CFG
-				    PE06_MSCR_S32R45 PE06_GMAC0_TX_D2_CFG
-				    PE07_MSCR_S32R45 PE07_GMAC0_TX_D3_CFG
-				    PE08_MSCR_S32R45 PE08_GMAC0_RX_CLK_CFG
-				    PE09_MSCR_S32R45 PE09_GMAC0_RX_DV_CFG
-				    PE10_MSCR_S32R45 PE10_GMAC0_RX_D0_CFG
-				    PE11_MSCR_S32R45 PE11_GMAC0_RX_D1_CFG
-				    PE12_MSCR_S32R45 PE12_GMAC0_RX_D2_CFG
-				    PE13_MSCR_S32R45 PE13_GMAC0_RX_D3_CFG
-				    GMAC0_TX_CLK_IMCR PE02_GMAC0_TX_CLK_IN
-				    GMAC0_RX_CLK_IMCR PE08_GMAC0_RX_CLK_IN
-				    GMAC0_RX_DV_IMCR PE09_GMAC0_RX_DV_IN
-				    GMAC0_RX_D0_IMCR PE10_GMAC0_RX_D0_IN
-				    GMAC0_RX_D1_IMCR PE11_GMAC0_RX_D1_IN
-				    GMAC0_RX_D2_IMCR PE12_GMAC0_RX_D2_IN
-				    GMAC0_RX_D3_IMCR PE13_GMAC0_RX_D3_IN
-				    >;
+		gmac0_grp0 {
+			pinmux = <S32CC_PINMUX(66, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-up;
+		};
+
+		gmac0_grp1 {
+			pinmux = <S32CC_PINMUX(67, FUNC1)>,
+				 <S32CC_PINMUX(68, FUNC1)>,
+				 <S32CC_PINMUX(69, FUNC1)>,
+				 <S32CC_PINMUX(70, FUNC1)>,
+				 <S32CC_PINMUX(71, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		gmac0_grp2 {
+			pinmux = <S32CC_PINMUX(72, FUNC1)>,
+				 <S32CC_PINMUX(73, FUNC1)>,
+				 <S32CC_PINMUX(74, FUNC1)>,
+				 <S32CC_PINMUX(75, FUNC1)>,
+				 <S32CC_PINMUX(76, FUNC1)>,
+				 <S32CC_PINMUX(77, FUNC1)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		gmac0_grp3 {
+			pinmux = <S32CC_PINMUX(538, FUNC2)>,
+				 <S32CC_PINMUX(529, FUNC2)>,
+				 <S32CC_PINMUX(530, FUNC2)>,
+				 <S32CC_PINMUX(531, FUNC2)>,
+				 <S32CC_PINMUX(532, FUNC2)>,
+				 <S32CC_PINMUX(533, FUNC2)>,
+				 <S32CC_PINMUX(534, FUNC2)>;
 		};
+
 	};
 
 	i2c0_pins: i2c0 {
-		i2c0_grp {
-			fsl,pins = <PB00_MSCR_S32R45 PB00_I2C0_SDA_CFG
-				    PB01_MSCR_S32R45 PB01_I2C0_SCL_CFG
-				    I2C0_SDA_IMCR PB00_I2C0_SDA_IN
-				    I2C0_SCL_IMCR PB01_I2C0_SCL_IN
-				    >;
+		i2c0_grp0 {
+			pinmux = <S32CC_PINMUX(16, FUNC1)>,
+				 <S32CC_PINMUX(17, FUNC1)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c0_grp1 {
+			pinmux = <S32CC_PINMUX(565, FUNC2)>,
+				 <S32CC_PINMUX(566, FUNC2)>;
 		};
+
 	};
 
 	i2c0_gpio_pins: i2c0_gpio {
-		i2c0_gpio_grp {
-			fsl,pins = <PB00_MSCR_S32R45 PB00_I2C0_SDA_GPIO
-				    PB01_MSCR_S32R45 PB01_I2C0_SCL_GPIO
-				    I2C0_SDA_IMCR IMCR_DISABLED
-				    I2C0_SCL_IMCR IMCR_DISABLED
-				    >;
+		i2c0_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(16, FUNC0)>,
+				 <S32CC_PINMUX(17, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
 		};
+
+		i2c0_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(565, FUNC0)>,
+				 <S32CC_PINMUX(566, FUNC0)>;
+		};
+
 	};
 
 	i2c1_pins: i2c1 {
-		i2c1_grp {
-			fsl,pins = <PA14_MSCR_S32R45 PA14_I2C1_SCL_CFG
-				    PA15_MSCR_S32R45 PA15_I2C1_SDA_CFG
-				    I2C1_SCL_IMCR PA14_I2C1_SCL_IN
-				    I2C1_SDA_IMCR PA15_I2C1_SDA_IN
-				    >;
+		i2c1_grp0 {
+			pinmux = <S32CC_PINMUX(14, FUNC4)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
 		};
+
+		i2c1_grp1 {
+			pinmux = <S32CC_PINMUX(15, FUNC3)>;
+			drive-open-drain;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		i2c1_grp2 {
+			pinmux = <S32CC_PINMUX(615, FUNC2)>,
+				 <S32CC_PINMUX(616, FUNC2)>;
+		};
+
 	};
 
 	i2c1_gpio_pins: i2c1_gpio {
-		i2c1_gpio_grp {
-			fsl,pins = <PA14_MSCR_S32R45 PA14_I2C1_SCL_GPIO
-				    PA15_MSCR_S32R45 PA15_I2C1_SDA_GPIO
-				    I2C1_SCL_IMCR IMCR_DISABLED
-				    I2C1_SDA_IMCR IMCR_DISABLED
-				    >;
+		i2c1_gpio_grp0 {
+			pinmux = <S32CC_PINMUX(14, FUNC0)>,
+				 <S32CC_PINMUX(15, FUNC0)>;
+			drive-open-drain;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		i2c1_gpio_grp1 {
+			pinmux = <S32CC_PINMUX(615, FUNC0)>,
+				 <S32CC_PINMUX(616, FUNC0)>;
 		};
+
 	};
 
 	qspi_pins: qspi {
-		qspi_grp {
-			fsl,pins = <PF05_MSCR_S32R45 PF05_QSPI_DATA_A0_CFG
-				    PF06_MSCR_S32R45 PF06_QSPI_DATA_A1_CFG
-				    PF07_MSCR_S32R45 PF07_QSPI_DATA_A2_CFG
-				    PF08_MSCR_S32R45 PF08_QSPI_DATA_A3_CFG
-				    PF09_MSCR_S32R45 PF09_QSPI_DATA_A4_CFG
-				    PF10_MSCR_S32R45 PF10_QSPI_DATA_A5_CFG
-				    PF11_MSCR_S32R45 PF11_QSPI_DATA_A6_CFG
-				    PF12_MSCR_S32R45 PF12_QSPI_DATA_A7_CFG
-				    PF13_MSCR_S32R45 PF13_QSPI_DQS_A_CFG
-				    PG00_MSCR_S32R45 PG00_QSPI_CLK_A_CFG
-				    PG01_MSCR_S32R45 PG01_QSPI_CLK_A_b_CFG
-				    PG02_MSCR_S32R45 PG02_QSPI_CLK_2A_CFG
-				    PG03_MSCR_S32R45 PG03_QSPI_CLK_2A_b_CFG
-				    PG04_MSCR_S32R45 PG04_QSPI_CS_A0
-				    PG05_MSCR_S32R45 PG05_QSPI_CS_A1
-				    QSPI_DATA0_A_IMCR PF05_QSPI_DATA_A0_IN
-				    QSPI_DATA1_A_IMCR PF06_QSPI_DATA_A1_IN
-				    QSPI_DATA2_A_IMCR PF07_QSPI_DATA_A2_IN
-				    QSPI_DATA3_A_IMCR PF08_QSPI_DATA_A3_IN
-				    QSPI_DATA4_A_IMCR PF09_QSPI_DATA_A4_IN
-				    QSPI_DATA5_A_IMCR PF10_QSPI_DATA_A5_IN
-				    QSPI_DATA6_A_IMCR PF11_QSPI_DATA_A6_IN
-				    QSPI_DATA7_A_IMCR PF12_QSPI_DATA_A7_IN
-				    QSPI_DQS_A_IMCR PF13_DQS_A_IN
-				    >;
+		qspi_grp0 {
+			pinmux = <S32CC_PINMUX(85, FUNC1)>,
+				 <S32CC_PINMUX(86, FUNC1)>,
+				 <S32CC_PINMUX(87, FUNC1)>,
+				 <S32CC_PINMUX(88, FUNC1)>,
+				 <S32CC_PINMUX(89, FUNC1)>,
+				 <S32CC_PINMUX(90, FUNC1)>,
+				 <S32CC_PINMUX(91, FUNC1)>,
+				 <S32CC_PINMUX(92, FUNC1)>,
+				 <S32CC_PINMUX(93, FUNC1)>;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-down;
+		};
+
+		qspi_grp1 {
+			pinmux = <S32CC_PINMUX(96, FUNC1)>,
+				 <S32CC_PINMUX(97, FUNC1)>,
+				 <S32CC_PINMUX(98, FUNC1)>,
+				 <S32CC_PINMUX(99, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		qspi_grp2 {
+			pinmux = <S32CC_PINMUX(100, FUNC1)>,
+				 <S32CC_PINMUX(101, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-up;
 		};
+
+		qspi_grp3 {
+			pinmux = <S32CC_PINMUX(540, FUNC2)>,
+				 <S32CC_PINMUX(541, FUNC2)>,
+				 <S32CC_PINMUX(542, FUNC2)>,
+				 <S32CC_PINMUX(543, FUNC2)>,
+				 <S32CC_PINMUX(544, FUNC2)>,
+				 <S32CC_PINMUX(545, FUNC2)>,
+				 <S32CC_PINMUX(546, FUNC2)>,
+				 <S32CC_PINMUX(547, FUNC2)>,
+				 <S32CC_PINMUX(548, FUNC2)>;
+		};
+
 	};
 
 	sd0_pins: sd0 {
-		sd0_grp {
-			fsl,pins = <PC14_MSCR_S32R45 PC14_SD0_CLK_CFG
-				    PC15_MSCR_S32R45 PC15_SD0_CMD_CFG
-				    PD00_MSCR_S32R45 PD00_SD0_D0_CFG
-				    PD01_MSCR_S32R45 PD01_SD0_D1_CFG
-				    PD02_MSCR_S32R45 PD02_SD0_D2_CFG
-				    PD03_MSCR_S32R45 PD03_SD0_D3_CFG
-				    PD04_MSCR_S32R45 PD04_SD0_D4_CFG
-				    PD05_MSCR_S32R45 PD05_SD0_D5_CFG
-				    PD06_MSCR_S32R45 PD06_SD0_D6_CFG
-				    PD07_MSCR_S32R45 PD07_SD0_D7_CFG
-				    PD08_MSCR_S32R45 PD08_SD0_RST_CFG
-				    PD09_MSCR_S32R45 PD09_SD0_VSELECT_CFG
-				    SD0_CMD_IMCR PC15_SD0_CMD_IN
-				    SD0_D0_IMCR PD00_SD0_D0_IN
-				    SD0_D1_IMCR PD01_SD0_D1_IN
-				    SD0_D2_IMCR PD02_SD0_D2_IN
-				    SD0_D3_IMCR PD03_SD0_D3_IN
-				    SD0_D4_IMCR PD04_SD0_D4_IN
-				    SD0_D5_IMCR PD05_SD0_D5_IN
-				    SD0_D6_IMCR PD06_SD0_D6_IN
-				    SD0_D7_IMCR PD07_SD0_D7_IN
-				    >;
+		sd0_grp0 {
+			pinmux = <S32CC_PINMUX(46, FUNC1)>,
+				 <S32CC_PINMUX(56, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-down;
+		};
+
+		sd0_grp1 {
+			pinmux = <S32CC_PINMUX(47, FUNC1)>,
+				 <S32CC_PINMUX(48, FUNC1)>,
+				 <S32CC_PINMUX(49, FUNC1)>,
+				 <S32CC_PINMUX(50, FUNC1)>,
+				 <S32CC_PINMUX(51, FUNC1)>,
+				 <S32CC_PINMUX(52, FUNC1)>,
+				 <S32CC_PINMUX(53, FUNC1)>,
+				 <S32CC_PINMUX(54, FUNC1)>,
+				 <S32CC_PINMUX(55, FUNC1)>;
+			output-enable;
+			input-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+			bias-pull-up;
+		};
+
+		sd0_grp2 {
+			pinmux = <S32CC_PINMUX(57, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_208MHZ>;
+		};
+
+		sd0_grp3 {
+			pinmux = <S32CC_PINMUX(515, FUNC2)>,
+				 <S32CC_PINMUX(516, FUNC2)>,
+				 <S32CC_PINMUX(517, FUNC2)>,
+				 <S32CC_PINMUX(520, FUNC2)>,
+				 <S32CC_PINMUX(521, FUNC2)>,
+				 <S32CC_PINMUX(522, FUNC2)>,
+				 <S32CC_PINMUX(523, FUNC2)>,
+				 <S32CC_PINMUX(519, FUNC2)>,
+				 <S32CC_PINMUX(518, FUNC2)>;
 		};
+
 	};
 
 	uart0_pins: uart0 {
-		uart0_grp {
-			u-boot,dm-pre-reloc;
-			fsl,pins = <PC09_MSCR_S32R45 PC09_LIN0_TX_CFG
-				    PC10_MSCR_S32R45 PC10_LIN0_RX_CFG
-				    LIN0_RX_IMCR PC10_LIN0_RX_IN>;
+		uart0_grp0 {
+			pinmux = <S32CC_PINMUX(41, FUNC1)>;
+			output-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
+		};
+
+		uart0_grp1 {
+			pinmux = <S32CC_PINMUX(42, FUNC0)>;
+			input-enable;
+			slew-rate = <S32CC_SLEW_83MHZ>;
 		};
-	};
 	};
 };
 
-- 
2.17.1

